slsa_for_models/gcp/README.md

```diff
@@ -0,0 +1,230 @@
+      kubectl patch configmap chains-config -n tekton-chains -p='{"data":{
+      "transparency.enabled": "true",
+      "artifacts.taskrun.format":"slsa/v2alpha2", 
+      "artifacts.taskrun.storage": "tekton",
+      "artifacts.pipelinerun.format":"slsa/v2alpha2", 
+      "artifacts.pipelinerun.storage": "tekton"
+      }}'
+      ```
+   4. Then restart the Tekton Chains controller to ensure it picks up the changes:
+
+      ```shell
+      kubectl delete pods -n tekton-chains -l app=tekton-chains-controller 
+      ```    
+
+7. Generate an encrypted x509 keypair and save it as a Kubernetes secret:
+   
+   ```shell
+   cosign generate-key-pair k8s://tekton-chains/signing-secrets
+   ```
+ 
+8. Apply the `Tasks`:
+
+   ```shell
+   kubectl apply -f slsa_for_models/gcp/tasks
+   ```
+
+9. Apply the `Pipeline`:
+
+   ```shell
+   kubectl apply -f slsa_for_models/gcp/pipeline.yml
+   ```
+ 
+10. Create a generic repository in Artifact Registry:
+
+    ```shell
+    gcloud artifacts repositories create <REPOSITORY> \
+     --location=<LOCATION> \
+     --repository-format=generic  
+    ```   
+   
+    For example:
+    ```shell
+    gcloud artifacts repositories create ml-artifacts \
+     --location=us \
+     --repository-format=generic  
+    ```
+   
+    If you set a different repository name and [location] from the example above, make sure to modify the `Parameter` 
+    named 'model-storage' in the `PipelineRun` with your own values. 
+
+11. Execute the `PipelineRun`:
+
+    ```shell
+    kubectl create -f slsa_for_models/gcp/pipelinerun.yml
+    ```
+
+12. Observe the `PipelineRun` execution:
+
+    ```shell
+    tkn pipelinerun logs --last --follow
+    ``` 
+
+13. When the `PipelineRun` succeeds, view its status:
+
+    ```shell
+    export PIPELINERUN_NAME=$(tkn pr describe --last -o  jsonpath='{.metadata.name}') 
+    kubectl get pipelinerun $PIPELINERUN_NAME -o yaml
+    ```
+
+14. View the SLSA attestation:
+
+   ```shell
+   cat sig | tr -d '\n' | pbcopy
+   pbpaste | jq '.payload | @base64d | fromjson'
+   ```
+
+15. View the transparency log entry in the public [Rekor][rekor] instance:
+
+   ```shell
+   export TLOG_ENTRY=$(tkn pr describe --last -o jsonpath="{.metadata.annotations.chains\.tekton\.dev/transparency}")
+   open $TLOG_ENTRY
+   ```
+   
+16. Retrieve the signature and payload from the `PipelinRun` which are stored as base64-encoded annotations:
+
+   ```shell
+   export PIPELINERUN_UID=$(tkn pr describe --last -o  jsonpath='{.metadata.uid}')
+   tkn pr describe --last -o jsonpath="{.metadata.annotations.chains\.tekton\.dev/signature-pipelinerun-$PIPELINERUN_UID}" | base64 -d > sig
+   ```
+
+17. View the attestation:
+
+   ```shell
+   cat sig | tr -d '\n' | pbcopy
+   pbpaste | jq '.payload | @base64d | fromjson'
+   ```
+
+18. Verify the attestation:
+
+   ```shell
+   cosign verify-blob-attestation --key k8s://tekton-chains/signing-secrets --signature sig --type slsaprovenance1 --check-claims=false /dev/null
```

Comment:
I'm not following this verification command. `sig` contains a signature only? The payload is set to `/dev/null` and you're passing `--check-claims=false`. So my understanding is that this it is verifying that a Sigstore-issued certificate signed something, but not what? `--check-claims=false` removes the verification of the provenance's intoto `Subject.Digest`, so the hash is not verified either?

Reply:
I'm not that familiar with cosign, was just following the commands in the Tekton Chains tutorial which gave that command for verification: https://tekton.dev/docs/chains/getting-started-tutorial/#configuring-tekton-chains

Let me dig into the details and update the verification check

Reply:
Thanks. `sig` seems to contain the DSSE https://github.com/secure-systems-lab/dsse. SO the verification is verifying the signature and the hash of the attestation, I think. But we should give it the model blob as input (instead of `dev/null`) and remove the `--check-claims=false` argument.

Reply:
Updated the cosign command to:

```shell
cosign verify-blob-attestation \
    --key k8s://tekton-chains/signing-secrets \
    --signature pytorch_model.pth.intoto.jsonl \
    --type slsaprovenance1 \
    pytorch_model.pth
```

Also planning to update Cosign commands in the Tekton Chains tutorials 
 
Thank you @laurentsimon 🙏🏾